Build fails if git shows signatures

## Bug Report

**Describe the bug**

We store and build fluentbit internally, storing it with signed commits. The cmake fails to extract the `FLB_GIT_HASH` correctly when the signature is shown, i.e. having `git config --global log.showSignature true`

**To Reproduce**

```
$ git log
commit a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2 (HEAD -> chore/v3.1.0, origin/chore/v3.1.0)
gpg: Signature made tue  9 jul 2024 08:39:15 CEST
gpg:                using RSA key A1B2C3D4E5F6A1B2C3D4E5F6A1B2C3D4E5F6A1B2
gpg: Good signature from "XL-SEC <xl-sec@xl-sec.se>" [ultimate]
Primary key fingerprint: E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
     Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
Author: XL-SEC <xl-sec@xl-sec.se>
Date:   Tue Jul 9 08:39:15 2024 +0200

    chore: Bump to version v3.1.0
(...)
$ cd build
$ cmake -DFLB_RELEASE=On -DFLB_JEMALLOC=On -DFLB_TLS=On -DFLB_SHARED_LIB=Off -DFLB_EXAMPLES=Off -DFLB_HTTP_SERVER=On -DFLB_IN_EXEC=Off -DFLB_IN_SYSTEMD=On -DFLB_OUT_KAFKA=On -DFLB_OUT_PGSQL=On -DFLB_NIGHTLY_BUILD="$FLB_NIGHTLY_BUILD" -DFLB_LOG_NO_CONTROL_CHARS=On -DFLB_CHUNK_TRACE="$FLB_CHUNK_TRACE" -DFLB_CONFIG_YAML=Off ..
(...)
$ grep -C3 FLB_GIT_HASH ../include/fluent-bit/flb_version.h
                             FLB_VERSION_MINOR * 100   \
                             FLB_VERSION_PATCH)
#define FLB_VERSION_STR     "3.1.0"
#define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
gpg:                using RSA key A1B2C3D4E5F6A1B2C3D4E5F6A1B2C3D4E5F6A1B2
gpg: Good signature from "XL-SEC <xl-sec@xl-sec.se>" [ultimate]
Primary key fingerprint: E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
(...)
$ make -j 13
(...)
[ 48%] Built target sqlite3
[ 48%] Building C object src/http_server/api/v2/CMakeFiles/api-v2.dir/reload.c.o
[ 48%] Building C object src/http_server/api/v2/CMakeFiles/api-v2.dir/__/__/__/__/lib/lwrb/lwrb/src/lwrb/lwrb.c.o
[ 48%] Building C object src/http_server/api/v2/CMakeFiles/api-v2.dir/metrics.c.o
[ 50%] Building C object src/http_server/api/v2/CMakeFiles/api-v2.dir/register.c.o
In file included from /home/xl-sec/git/images/fluent-bit/src/src/http_server/api/v2/metrics.c:26:
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: warning: missing terminating " character
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:43:4: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘:’ token
   43 | gpg:                using RSA key A1B2C3D4E5F6A1B2C3D4E5F6A1B2C3D4E5F6A1B2
      |    ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:31: error: invalid suffix "E5F" on integer constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                               ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:36: error: invalid suffix "A" on floating constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:67: error: invalid suffix "E5F" on integer constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                                                                   ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:46:36: error: invalid suffix "E" on integer constant
   46 |      Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
      |                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:46:52: error: invalid suffix "E5F" on integer constant
   46 |      Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
      |                                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:1: error: invalid suffix "f5ecb53d0b584bf1e6bb83b59bc5c98f8522a9c" on integer constant
   47 | a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:41: warning: missing terminating " character
   47 | a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
      |                                         ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:41: error: missing terminating " character
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: error: missing terminating " character
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: note: in definition of macro ‘FLB_GIT_HASH’
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/xl-sec/git/images/fluent-bit/src/src/http_server/api/v2/reload.c:26:
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: warning: missing terminating " character
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:43:4: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘:’ token
   43 | gpg:                using RSA key A1B2C3D4E5F6A1B2C3D4E5F6A1B2C3D4E5F6A1B2
      |    ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:31: error: invalid suffix "E5F" on integer constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                               ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:36: error: invalid suffix "A" on floating constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:45:67: error: invalid suffix "E5F" on integer constant
   45 | Primary key fingerprint: 2100 E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
      |                                                                   ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:46:36: error: invalid suffix "E" on integer constant
   46 |      Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
      |                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:46:52: error: invalid suffix "E5F" on integer constant
   46 |      Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
      |                                                    ^~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:1: error: invalid suffix "f5ecb53d0b584bf1e6bb83b59bc5c98f8522a9c" on integer constant
   47 | a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:41: warning: missing terminating " character
   47 | a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
      |                                         ^
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:47:41: error: missing terminating " character
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: error: missing terminating " character
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/xl-sec/git/images/fluent-bit/src/include/fluent-bit/flb_version.h:42:29: note: in definition of macro ‘FLB_GIT_HASH’
   42 | #define FLB_GIT_HASH        "gpg: Signature made tue  9 jul 2024 08:39:15 CEST
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/http_server/api/v2/CMakeFiles/api-v2.dir/build.make:76: src/http_server/api/v2/CMakeFiles/api-v2.dir/metrics.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/http_server/api/v2/CMakeFiles/api-v2.dir/build.make:90: src/http_server/api/v2/CMakeFiles/api-v2.dir/reload.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:8589: src/http_server/api/v2/CMakeFiles/api-v2.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**Expected behavior**

```
$ git log
commit a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2 (HEAD -> chore/v3.1.0, origin/chore/v3.1.0)
gpg: Signature made tue  9 jul 2024 08:39:15 CEST
gpg:                using RSA key A1B2C3D4E5F6A1B2C3D4E5F6A1B2C3D4E5F6A1B2
gpg: Good signature from "XL-SEC <xl-sec@xl-sec.se>" [ultimate]
Primary key fingerprint: E5F6 A1B2 C3D4 E5F6 A1B2  A1B2 C3D4 E5F6 A1B2 C3D4
     Subkey fingerprint: A1B2 C3D4 E5F6 A1B2 C3D4  E5F6 A1B2 C3D4 E5F6 A1B2
Author: XL-SEC <xl-sec@xl-sec.se>
Date:   Tue Jul 9 08:39:15 2024 +0200

    chore: Bump to version v3.1.0
$ cd build
$ cmake -DFLB_RELEASE=On -DFLB_JEMALLOC=On -DFLB_TLS=On -DFLB_SHARED_LIB=Off -DFLB_EXAMPLES=Off -DFLB_HTTP_SERVER=On -DFLB_IN_EXEC=Off -DFLB_IN_SYSTEMD=On -DFLB_OUT_KAFKA=On -DFLB_OUT_PGSQL=On -DFLB_NIGHTLY_BUILD="$FLB_NIGHTLY_BUILD" -DFLB_LOG_NO_CONTROL_CHARS=On -DFLB_CHUNK_TRACE="$FLB_CHUNK_TRACE" -DFLB_CONFIG_YAML=Off ..
(...)
$ grep -C3 FLB_GIT_HASH ../include/fluent-bit/flb_version.h
                             FLB_VERSION_MINOR * 100   \
                             FLB_VERSION_PATCH)
#define FLB_VERSION_STR     "3.1.0"
#define FLB_GIT_HASH        "a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2"
(...)
$ make -j 13
(...)
[100%] Building C object src/CMakeFiles/fluent-bit-bin.dir/__/lib/lwrb/lwrb/src/lwrb/lwrb.c.o
[100%] Linking C executable ../bin/fluent-bit
[100%] Built target fluent-bit-bin
```

**Your Environment**

* Version used: v3.1.0
* Configuration: `git config --global log.showSignature true`

**Proposed fix**

Easiest is to include `-c log.showSignature=false` to the git command in `CMakeList.txt`. Unless anyone knows a fancier solution I can create a PR for that.

```
# Extract Git commit information for debug output.
# Note that this is only set when cmake is run, the intent here is to use in CI for verification of releases so is acceptable.
# For a better solution see https://jonathanhamberg.com/post/cmake-embedding-git-hash/ but this is simple and easy.
find_package(Git)
# If we do not have Git or this is not a Git repo or another error this just is ignored and we have no output at runtime.
execute_process(COMMAND
  "${GIT_EXECUTABLE}" -c log.showSignature=false log -1 --format=%H
  WORKING_DIRECTORY "${FLB_ROOT}"
  OUTPUT_VARIABLE FLB_GIT_HASH
  ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)
message(STATUS "Git hash: ${FLB_GIT_HASH}")
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -285,7 +285,7 @@ endif ()
 find_package(Git)
 # If we do not have Git or this is not a Git repo or another error this just is ignored and we have no output at runtime.
 execute_process(COMMAND
-  "${GIT_EXECUTABLE}" log -1 --format=%H
+  "${GIT_EXECUTABLE}" -c log.showSignature=false log -1 --format=%H
   WORKING_DIRECTORY "${FLB_ROOT}"
   OUTPUT_VARIABLE FLB_GIT_HASH
   ERROR_QUIET OUTPUT_STRIP_TRAILING_WHITESPACE)

